lib/finch.ex

```diff
@@ -50,6 +50,28 @@ defmodule Finch do
     ]
   ]
 
+  @typedoc """
+  The `:name` provided to Finch in `start_link/1`.
+  """
+  @type name() :: atom()
+
+  @typedoc """
+  An HTTP request method represented as an atom or a String in all caps.
+
+  The following methods are supported: #{Enum.join(@methods, ", ")}.
+  """
+  @type http_method() :: :get | :post | :head | :patch | :delete | :options | String.t()
+
+  @typedoc """
+  A Uniform Resource Locator, the address of a resource on the Web.
+  """
+  @type url() :: String.t()
+
+  @typedoc """
+  A body associated with a request.
+  """
+  @type body() :: binary() | nil
```

Comment:
Just curious, but doesn't Finch support iodata?

Reply:
Ah right, it certainly does, thanks!